lib/libspl/include/sys/uio.h

```diff
@@ -82,6 +82,32 @@ typedef struct zfs_uio {
 #define	zfs_uio_iovlen(uio, idx)	(uio)->uio_iov[(idx)].iov_len
 #define	zfs_uio_iovbase(uio, idx)	(uio)->uio_iov[(idx)].iov_base
 
+static inline boolean_t
+zfs_dio_page_aligned(void *buf)
+{
+	return ((((unsigned long)(buf) & (PAGESIZE - 1)) == 0) ?
+	    B_TRUE : B_FALSE);
+}
```

Comment:
I wound up doing a deep dive not this today, as I forgot why this was even the case of not putting `zfs_dio_page_aligned()` under a `_KERNEL` guard originally in `uio_impl.h`. Way back in the day, I updated `zfs_context.h` to account for including `uio_impl.h` for the kernel and `uio.h` for the user-space code. I just completely forgot about that. I went back and removed including `uio_impl.h` where it was not necessary in the latest commit. Also, there is user-space code in ztest that uses `zfs_dio_page_aligned()`. Calls to `dmu_read()` ->`dmu_read_impl()`. This just exercises part of the Direct I/O code for reads outside of the kernel pinning user pages. If you still believe there is places that are not covered by the guards though, please let me know.

Reply:
My original point was that user-space is not supposed to have compile-time defines for `PAGESIZE`, etc, since the same user-space binary may run on systems (of just kernel configs) with different page sizes.  If you look into `lib/libzpool/abd_os.c`, there are completely arbitrary defines for `ABD_PAGESIZE` and `ABD_PAGEMASK`.

Reply:
Ah okay. I am open to any suggestions on how to resolve the issue ASAP. Do you have an idea how to get this resolved in a beter way?

Reply:
I hoped your answer will be: "Sure, we don't need it, since we do not implement Direct I/O in user-space, lets just stub it."  But I see that you are randomly set `DMU_DIRECTIO` in few places `ztest`, so the question looks like: "what part of Direct I/O functionality we can and want to reasonably implement in user-space and how can we give it coherent notion of `PAGE*` macros in all places that really need it".

Reply:
I don't know how many platforms with PAGESIZE != 4096 in user-space do we have to cause troubles.  From ASAP perspective I guess it might be not a show-stopper, if we do not care about `ztest` there right now.